src/aggregation/count_distinct.c

```diff
@@ -550,8 +552,16 @@ static void count_distinct_noise_final_type(const ArgsDescriptor *args_desc, Oid
 
 static Datum count_distinct_noise_finalize(AnonAggState *base_state, Bucket *bucket, BucketDescriptor *bucket_desc, bool *is_null)
 {
-  CountDistinctResult result = count_distinct_calculate_final(base_state, bucket, bucket_desc, is_null);
-  return Float8GetDatum(result.noise_sd);
+  CountDistinctResult result = count_distinct_calculate_final(base_state, bucket, bucket_desc);
+  if (result.not_enough_aid_values && result.noisy_count == 0)
```

Comment:
Should this be an OR?

Reply:
ha, I actually got surprised by this 5 mins after writing it myself. This is the condition, where `count_distinct` not only doesn't have any low-count contributions (because `not_enough_aid_values`) but also doesn't have any high-count contributing values.

Since this is surprising to you as well, maybe I need to double down over it once more and refactor so that it is more clear.

Reply:
yup, this piece of logic is clearly misplaced, fixing...